tools/cli/workflowBatchCommands.go

```diff
@@ -202,13 +212,38 @@ func StartBatchJob(c *cli.Context) {
 		},
 		RPS: rps,
 	}
-	wf, err := client.StartWorkflow(tcCtx, options, batcher.BatchWFTypeName, params)
+	input, err := json.Marshal(params)
+	if err != nil {
+	}
+	workflowId := uuid.NewRandom().String()
+	request := &types.StartWorkflowExecutionRequest{
+		Domain:                              domain,
+		RequestID:                           uuid.New(),
+		WorkflowID:                          workflowId,
+		ExecutionStartToCloseTimeoutSeconds: common.Int32Ptr(int32(batcher.InfiniteDuration.Seconds())),
+		TaskList:                            &types.TaskList{Name: batcher.BatcherTaskListName},
+		Memo: &types.Memo{
+			Fields: map[string][]byte{
+				"Reason": []byte(reason),
+			},
+		},
+		SearchAttributes: &types.SearchAttributes{
+			IndexedFields: map[string][]byte{
+				"CustomDomain": []byte(domain),
+				"Operator":     []byte(operator),
+			},
+		},
+		WorkflowType: &types.WorkflowType{Name: batcher.BatchWFTypeName},
+		Input:        input,
+	}
+	// which domain does this run in?
```

Comment:
This should be `common.BatcherLocalDomainName`.
While we query for domains provided by user `--domain`, the actual workflow runs in `common.BatcherLocalDomainName`. Previous client used this when it was initialized at the start of the function, therefore we need to pass it in start request now.